### PR TITLE
Fix windows samba mount.

### PR DESCRIPTION
**Pull request by michael-molchanov**
Add right permisions on windows smb2 mount.
Add smb3 option for users, that want to create smb share manually and only once.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -145,10 +145,20 @@ Vagrant.configure("2") do |config|
     config.vm.provision "shell", run: "always" do |s|
       s.inline = <<-SCRIPT
         mkdir -p vagrant $2
-        mount -t cifs -o uid=`id -u docker`,gid=`id -g docker`,sec=ntlm,username=$3,pass=$4 //192.168.10.1/$1 $2
```
Comment (achekulaev) [inline Vagrantfile]: Why have you removed host IP in this line?

**Comment by lmakarov** [comment]
@michael-molchanov looks like the only difference between **smb2** and **smb3** option that you propose are the two triggers to create/delete the share upon vagrant up/destroy.
I'd rather have a flag for this in `vagrant.yml` (e.g. `smb2_auto: true/false`).

File: Vagrantfile (hunk 1 above)
Comment (lmakarov) [inline Vagrantfile:148]: @achekulaev What do you mean? `//192.168.10.1/$1` is still there :)

File: vagrant.yml [hunk 2]
```diff
@@ -12,6 +12,8 @@ synced_folders:
     # If using the 'smb2' type above the user and share will be configured automatically.
     smb_username: 'vagrant'
     smb_password: 'vagrant'
+    # Create smb share on vagrant up. Set to false if you already have smb shared folder.
```
Comment (lmakarov) [inline vagrant.yml]: Please change to "Automatically provision the user account and smb share on vagrant up. Set to false if you want to or already created the user and the share manually."

Also, it is not obvious what the share name should be in order for this to work. So worth adding a note here as well.

File: vagrant.yml [hunk 3]
```diff
@@ -12,6 +12,8 @@ synced_folders:
...
+    smb_create: true
```
Comment (lmakarov) [inline vagrant.yml]: Please name this "smb2_auto", since it's related strictly to smb2, not smb in general.

**Pull request merged**